### ISSUE TITLE
Help: trying to set up a simple zsh autostart plugin for zellij

### ISSUE THREAD

**Issue by tranzystorek-io** [zellij-org/zellij#904]
I wanted to make an autostart zsh plugin that would start zellij automatically and close the terminal once I close the shell,
inspired by https://github.com/zpm-zsh/tmux

My `zellij.plugin.zsh` so far:

```zsh
#!/usr/bin/env zsh

if (( $+commands[zellij] )); then
    function _zellij_autostart() {
        zellij -s main
        exit 0
    }

    autoload -Uz add-zsh-hook
    add-zsh-hook precmd _zellij_autostart
fi
```

When I add this to a plugin manager like zplug, zellij starts for ~1s and then closes. My suspicion is that zellij starts its server and then the command terminates.

I also tried switching the command to `zellij attach -c main` but then zellij goes crazy (flickering interface, not responding to keyboard input).

I'm wondering if it's an easy fix or if there is some fundamental difference between how zellij and tmux works?

**Comment by a-kenji** [comment]
Hey, thanks for opening the issue. I haven't had time to really look at it yet, but maybe this helps:

If there is already a session open. Zellij doesn't  let you create a new one with `-s`

If you want to attach to a running session, and create one, if there is no session running already you can do so with `zellij attach -c main`.

I hope this helps, if not please reach out again!

**Comment by a-kenji** [comment]
@tranzystorek-io 

Maybe you are stuck in a loop, since that plugin might be loaded more than once?

I am not familiar with zsh, but something like that could alleviate this behavior for bash:

```
#!/usr/bin/env bash

If [[ -v ZELLIJ ]]
then
       true
else
       zellij attach -c main
fi
```

**Comment by tranzystorek-io** [comment]
Unfortunately adding a `[[ ! -v ZELLIJ ]]` check didn't change anything.

I think the loop might be the case though:

![obraz](https://user-images.githubusercontent.com/5671049/143682387-f7d9eb72-6473-488d-a251-b175f38362fe.png)

**Comment by tranzystorek-io** [comment]
Update: this problem only appears when I'm using the cargo version, when using the package from arch repository, everything is fine

**Comment by tranzystorek-io** [comment]
Update 2: just reproduced the faulty behavior by running `zellij attach` from within a running zellij session

**Comment by a-kenji** [comment]
Thank you for finding that @tranzystorek-io,
Ideally zellij should warn here that you are trying to connect to the 
session you are already in. That definitely is a bug.

I still find it slightly weird that the plugin doesn't see the `ZELLIJ` env var. That zellij exposes.

**Comment by tranzystorek-io** [comment]
To be exact, my final form of the plugin script does use the ZELLIJ variable:

```zsh
#!/usr/bin/env zsh

_is_zellij_cmd_available() {
    [[ -v commands[zellij] ]]
}

_is_zellij_running() {
    [[ -v ZELLIJ ]]
}

if _is_zellij_cmd_available && ! _is_zellij_running; then
    function _zellij_autostart() {
        exec zellij
    }

    autoload -Uz add-zsh-hook
    add-zsh-hook precmd _zellij_autostart
fi
```

**Comment by a-kenji** [comment]
@tranzystorek-io,
Does that work for you, or do you still have an issue with it?

**Comment by tranzystorek-io** [comment]
It works fine, except for the `zellij attach`-in-running-zellij-window bug I mentioned

**Comment by a-kenji** [comment]
Thank you for that information! Also thanks for sharing that zsh plugin, that is pretty awesome.

**Comment by tranzystorek-io** [comment]
I published the repo in case anyone wants to use it: https://github.com/tranzystorek-io/zellij-zsh

**Comment by a-kenji** [comment]
The attaching issue has been addressed in  #911.
Please don't hesitate reaching out in the future!
